src/vs/vscode.proposed.d.ts

```diff
@@ -33,7 +33,31 @@ declare module 'vscode' {
 		extensionHostEnv?: { [key: string]: string | null };
 	}
 
-	export type ResolverResult = ResolvedAuthority & ResolvedOptions;
+	export interface TunnelOptions {
+		remote: { port: number, host: string };
+		localPort?: number;
+		name?: string;
+		closeable?: boolean;
+	}
+
+	export interface Tunnel extends Disposable {
+		remote: { port: number, host: string };
+		local: { port: number, host: string };
+	}
+
+	/**
+	 * Used as part of the ResolverResult if the extension has any candidate, published, or forwarded ports.
+	 */
+	export interface TunnelInformation {
+		/**
+		 * Tunnels that are detected by the extension. The remotePort is used for display purposes.
+		 * The localAddress should be the complete local address for connecting to the port. Tunnels provided through
+		 * detected are read-only from the forwarded ports UI.
+		 */
+		detected?: { remotePort: number, localAddress: string }[];
```

Comment:
`complete local address for connecting to the port` like a URL or just `localhost:1234`?

Reply:
just `localhost:1234`, I'll add an example to the comment.